File: z-clients/javascript/src/models/adminAuthTokenCreateOut.ts
// this file is @generated

export interface AdminAuthTokenCreateOut {
    id: string;
    token: string;
    created: Date;
    updated: Date;
}

export const AdminAuthTokenCreateOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AdminAuthTokenCreateOut {
        return {
            id: object['id'],
            token: object['token'],
            created: new Date(Number(object['created'])),
            updated: new Date(Number(object['updated'])),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AdminAuthTokenCreateOut): any {
        return {
            'id': self.id,
            'token': self.token,
            'created': self.created.getTime(),
            'updated': self.updated.getTime(),
        };
    }
}

File: z-clients/rust/src/models/admin_auth_token_create_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AdminAuthTokenCreateOut {
    pub id: String,

    pub token: String,

    pub created: u64,

    pub updated: u64,
}

impl AdminAuthTokenCreateOut {
    pub fn new(id: String, token: String, created: u64, updated: u64) -> Self {
        Self {
            id,
            token,
            created,
            updated,
        }
    }
}
